eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,183 @@
+/*
+ *
+ *  * Copyright 2016 EMBL - European Bioinformatics Institute
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *      http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package embl.ebi.variation.eva.vcfdump;
+
+import embl.ebi.variation.eva.vcfdump.cellbasewsclient.CellbaseWSClient;
+import embl.ebi.variation.eva.vcfdump.exception.CellbaseSequenceDownloadError;
+import htsjdk.variant.variantcontext.*;
+import org.broadinstitute.variant.vcf.VCFConstants;
+import org.opencb.biodata.models.feature.Region;
+import org.opencb.biodata.models.variant.Variant;
+import org.opencb.biodata.models.variant.VariantSource;
+import org.opencb.biodata.models.variant.VariantSourceEntry;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Created by pagarcia on 27/06/2016.
+ */
+public class BiodataVariantToVariantContextConverter {
+
+    public static final String GENOTYPE_KEY = "GT";
+    private final VariantContextBuilder variantContextBuilder;
+    private final List<VariantSource> sources;
+    private CellbaseWSClient cellbaseClient;
+    private String regionSequence;
+    private Map<String, Map<String,String>> studiesSampleNamesEquivalences;
+    private static final int NO_CALL_ALLELE_INDEX = 2;
+
+    public BiodataVariantToVariantContextConverter(List<VariantSource> sources, CellbaseWSClient cellbaseWSClient,
+                                                   Map<String, Map<String,String>> studiesSampleNamesEquivalences)
+    {
+        this.sources = sources;
+        this.cellbaseClient = cellbaseWSClient;
+        this.studiesSampleNamesEquivalences = studiesSampleNamesEquivalences;
+        variantContextBuilder = new VariantContextBuilder();
+    }
+
+    public VariantContext transform(Variant variant, Region region) throws CellbaseSequenceDownloadError {
+        String[] allelesArray = getAllelesArray(variant, region);
+
+        Set<Genotype> genotypes = getGenotypes(variant, allelesArray);
+
+        VariantContext variantContext = variantContextBuilder
+                .chr(variant.getChromosome())
+                // TODO: check start and end for indels
```

Comment:
Ok, need to fix that. This deletion in the submitted file:
`21      820722  ss1098874278    TACC    T       .`
is being downloaded as 
`21	820723	.	TACC	T`
And the same with 1 nt deletions: 
`21      824457  ss1098874287    TA      T
`to
`21	824458	.	TA	T `
Same issue with insertions. From:
`21	822427	.	C	CT	.` (submitted file)
to
`21	822428	.	C	CT` (downloaded file)